backend/app/api/opentripmap.py

```diff
@@ -0,0 +1,184 @@
+"""
+This script contains the logic for interacting with the OpenTripMap API using FastAPI.
+(https://rapidapi.com/opentripmap/api/places1/playground/apiendpoint_c92ce138-db30-4713-9058-752835430bc7)
+It can be seen as an extension of the events.py file.
+
+Author: Ruben Vandamme
+Date: 04 November 2024
+Usage:
+    - Call the query_opentripmap function with a name of a city to get a list of relevant Events.
+"""
+
+import requests
+from typing import List
+from app.models.event_model import Event
+from pathlib import Path
+from app.config import OPENTRIPMAP_API_KEY
+
+HOST = "opentripmap-places-v1.p.rapidapi.com"
+
+def make_request(endpoint: str, query_params: dict) -> dict:
+    """General request function for OpenTripMap API"""
+
+    # variables
+    headers = {
+        "X-RapidAPI-Key": OPENTRIPMAP_API_KEY,
+        "X-RapidAPI-Host": HOST
+    }
+    full_url = f"https://{HOST}/{endpoint}"
+
+    # make request
+    response = requests.get(
+        url=full_url, 
+        headers=headers, 
+        params=query_params
+    )
+
+    # parse request if valid
+    if response.status_code == 200:
+        return response.json()
+    else:
+        raise Exception(f"""
+            Error {response.status_code} in make_request 
+            for url {full_url} 
+            with headers {headers} 
+            and params {query_params}
+            : {response.text}
+        """)
+
+valid_kinds_file_path = Path(__file__).resolve().parents[2] / 'app' / 'api' / 'opentripmap_valid_kinds.txt'
+
+def query_opentripmap(
+    placename:      str,     
```

Comment:
it's asking itself to use pydantic here and validate the fields https://docs.pydantic.dev/latest/ https://docs.pydantic.dev/latest/concepts/validators/